cpp/models/abm/location.cpp

```diff
@@ -43,7 +43,7 @@ Location::Location(LocationType type, uint32_t index, uint32_t num_cells)
     , m_npi_active(false)
 {
     // Initialize the first time point and set the subpopulation values to 0. 
-    m_subpopulations.add_time_point();
+    m_subpopulations.add_time_point(0);
```

Comment:
Yes, I also think that 0 is not a good one but choose a simple path to avoid adding a parameter to the constructor.
I can also adjust this if needed.

Reply:
I think it is better to not have it initialized

Reply:
The function 'add_person', which is called when we set up the world/simulation, calls 'change_subpopulation'. This function relies on an initialized subpopulation time series. I don't know if we can skip this or maybe delete this Timepoint later on.

Reply:
How about letting it unitialized and setting it here https://github.com/DLR-SC/memilio/blob/main/cpp/models/abm/simulation.cpp#L53 ? 

This is not the final and best idea as this may lead to discrepancies in values stored here and there... Could you try to elaborate a solution @DavidKerkmann ?

Reply:
From the discussion, I adapted to a (not elegant) solution: 
1) Create a function to initialize the subpopulation.
2) Use it when we initialise the location.  
3) The subpopulation is then reinitialized to the first Timepoint of the simulation.  

I can then create an issue which later can use David's vector of current persons saved in the location to replace the (2) point. Let me know what you think @mknaranja  @DavidKerkmann ?